com.microsoft.mrtk.core/Editor/MRTKBuildPreferences.cs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Settings provider for build-specific settings, like the 3D app launcher model for Windows builds.
+    /// </summary>
+    [Serializable]
+    internal class MRTKBuildPreferences : IPreprocessBuildWithReport, IPostprocessBuildWithReport
+    {
+        private const string AppLauncherPath = @"Assets\AppLauncherModel.glb";
+        private const string AppLauncherDocsUrl = @"https://docs.microsoft.com/windows/mixed-reality/distribute/creating-3d-models-for-use-in-the-windows-mixed-reality-home";
+
+        private static GUIContent appLauncherModelLabel = null;
+        private static GUIContent buttonContent = null;
+        private static UnityEditor.Editor gameObjectEditor = null;
+        private static GUIStyle appLauncherPreviewBackgroundColor = null;
+        private static bool isBuilding = false;
+
+        // Arbitrary callback order, chosen to be larger so that it runs after other things that
+        // a developer may have already.
+        int IOrderedCallback.callbackOrder => 100;
+
+        /// <remarks>See <see href="https://docs.microsoft.com/windows/mixed-reality/distribute/3d-app-launcher-design-guidance">3D app launcher design guidance</see> for more information.</remarks>
```

Comment:
I have not seen remarks doc comments w/o a summary before.... should we add one here? (i know it is private, but.... doesnt feel right)

Reply:
I mostly just wanted this info _somewhere_. I can adjust the location

Reply:
then feel free to merge, my friend :)

Reply:
Moved this to the class summary!